From b4ed9bb79c7911047294ba8ae081039927283dae Mon Sep 17 00:00:00 2001
From: Paulo Zanoni <pzanoni@mandriva.com>
Date: Thu, 6 Jan 2011 16:31:54 -0200
Subject: [PATCH] Mips support

---
 include/GL/internal/sarea.h |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/include/GL/internal/sarea.h b/include/GL/internal/sarea.h
index a0d6084..369f589 100644
--- a/include/GL/internal/sarea.h
+++ b/include/GL/internal/sarea.h
@@ -44,6 +44,8 @@
 #define SAREA_MAX 			0x2000
 #elif defined(__ia64__)
 #define SAREA_MAX			0x10000		/* 64kB */
+#elif defined(__mips__)
+#define SAREA_MAX			getpagesize()
 #else
 /* Intel 830M driver needs at least 8k SAREA */
 #define SAREA_MAX			0x2000
-- 
1.7.1

