From 7e397c0ae8bd6f42f06b3312a8db36ecb16c209e Mon Sep 17 00:00:00 2001
From: Boris Brezillon <boris.brezillon@collabora.com>
Date: Thu, 13 Apr 2023 14:47:30 +0200
Subject: [PATCH] ceu_branch

---
 src/panfrost/lib/genxml/ceu_builder.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/panfrost/lib/genxml/ceu_builder.h b/src/panfrost/lib/genxml/ceu_builder.h
index f763fc11848..f509c7cc517 100644
--- a/src/panfrost/lib/genxml/ceu_builder.h
+++ b/src/panfrost/lib/genxml/ceu_builder.h
@@ -285,6 +285,18 @@ ceu_wait_slots(ceu_builder *b, uint8_t slots)
    }
 }
 
+static inline void
+ceu_branch(ceu_builder *b, int16_t offset,
+           enum mali_ceu_condition cond, ceu_index val)
+{
+   ceu_emit(b, BRANCH, I)
+   {
+      I.offset = offset;
+      I.condition = cond;
+      I.value = ceu_to_reg32(val);
+   }
+}
+
 static inline void
 ceu_run_compute(ceu_builder *b, unsigned task_increment,
                 enum mali_task_axis task_axis)
-- 
GitLab

