diff -up mesa-23.2.0/src/panfrost/lib/pan_bo.c.11~ mesa-23.2.0/src/panfrost/lib/pan_bo.c
--- mesa-23.2.0/src/panfrost/lib/pan_bo.c.11~	2023-07-29 15:29:35.330000730 +0200
+++ mesa-23.2.0/src/panfrost/lib/pan_bo.c	2023-07-29 15:35:16.182963282 +0200
@@ -162,11 +162,9 @@ panfrost_bo_free(struct panfrost_bo *bo)
    int ret;
 
    if (bo->dev->arch >= 10) {
-      if (bo->sync.handle && bo->sync.point) {
-         ret = drmSyncobjTimelineWait(bo->dev->fd, &bo->sync.handle,
-                                      &bo->sync.point, 1, INT64_MAX,
-                                      DRM_SYNCOBJ_WAIT_FLAGS_WAIT_ALL, NULL);
-      }
+      /* Make sure all jobs are done accessing this buffer before unmapping. */
+      ret = panfrost_bo_wait(bo, INT64_MAX, true);
+      assert(ret);
 
       struct drm_pancsf_vm_unmap unmap = {
          .vm_id = bo->dev->vm_id,
@@ -180,7 +178,6 @@ panfrost_bo_free(struct panfrost_bo *bo)
       util_vma_heap_free(&dev->vma_heap, bo->ptr.gpu, bo->size);
 
       drmSyncobjDestroy(dev->fd, bo->sync.handle);
-
    }
 
    /* BO will be freed with the sparse array, but zero to indicate free */
