diff -up mesa-18.2.5/src/util/Makefile.am.0002~ mesa-18.2.5/src/util/Makefile.am
--- mesa-18.2.5/src/util/Makefile.am.0002~	2018-11-15 19:41:42.872730114 +0100
+++ mesa-18.2.5/src/util/Makefile.am	2018-11-15 19:42:24.560017567 +0100
@@ -60,7 +60,7 @@ libmesautil_la_LIBADD = \
 	$(CLOCK_LIB) \
 	$(ZLIB_LIBS) \
 	$(LIBATOMIC_LIBS) \
-	-lm
+	-lsupc++ -lm
 
 libxmlconfig_la_SOURCES = $(XMLCONFIG_FILES)
 libxmlconfig_la_CFLAGS = \
