diff -up mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/blend_jit.cpp.0207~ mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/blend_jit.cpp
--- mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/blend_jit.cpp.0207~	2018-07-06 23:12:00.000000000 +0200
+++ mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/blend_jit.cpp	2018-07-07 01:59:00.802228228 +0200
@@ -870,7 +870,6 @@ struct BlendJit : public Builder
         passes.add(createCFGSimplificationPass());
         passes.add(createEarlyCSEPass());
         passes.add(createInstructionCombiningPass());
-        passes.add(createInstructionSimplifierPass());
         passes.add(createConstantPropagationPass());
         passes.add(createSCCPPass());
         passes.add(createAggressiveDCEPass());
diff -up mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/fetch_jit.cpp.0207~ mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/fetch_jit.cpp
--- mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/fetch_jit.cpp.0207~	2018-07-06 23:12:00.000000000 +0200
+++ mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/fetch_jit.cpp	2018-07-07 01:59:00.803228228 +0200
@@ -294,7 +294,6 @@ Function* FetchJit::Create(const FETCH_C
     optPasses.add(createCFGSimplificationPass());
     optPasses.add(createEarlyCSEPass());
     optPasses.add(createInstructionCombiningPass());
-    optPasses.add(createInstructionSimplifierPass());
     optPasses.add(createConstantPropagationPass());
     optPasses.add(createSCCPPass());
     optPasses.add(createAggressiveDCEPass());
diff -up mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/functionpasses/lower_x86.cpp.0207~ mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/functionpasses/lower_x86.cpp
--- mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/functionpasses/lower_x86.cpp.0207~	2018-07-07 01:59:00.803228228 +0200
+++ mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/functionpasses/lower_x86.cpp	2018-07-07 02:00:26.961231441 +0200
@@ -76,7 +76,7 @@ namespace SwrJit
         {"meta.intrinsic.VCVTPS2PH", Intrinsic::x86_vcvtps2ph_256},
         {"meta.intrinsic.VPTESTC", Intrinsic::x86_avx_ptestc_256},
         {"meta.intrinsic.VPTESTZ", Intrinsic::x86_avx_ptestz_256},
-        {"meta.intrinsic.VFMADDPS", Intrinsic::x86_fma_vfmadd_ps_256},
+        {"meta.intrinsic.VFMADDPS", Intrinsic::fmuladd},
         {"meta.intrinsic.VPHADDD", Intrinsic::x86_avx2_phadd_d},
         {"meta.intrinsic.PDEP32", Intrinsic::x86_bmi_pdep_32},
         {"meta.intrinsic.RDTSC", Intrinsic::x86_rdtsc},
diff -up mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/streamout_jit.cpp.0207~ mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/streamout_jit.cpp
--- mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/streamout_jit.cpp.0207~	2018-07-06 23:12:00.000000000 +0200
+++ mesa-18.2-20180707/src/gallium/drivers/swr/rasterizer/jitter/streamout_jit.cpp	2018-07-07 01:59:00.803228228 +0200
@@ -306,7 +306,6 @@ struct StreamOutJit : public Builder
         passes.add(createCFGSimplificationPass());
         passes.add(createEarlyCSEPass());
         passes.add(createInstructionCombiningPass());
-        passes.add(createInstructionSimplifierPass());
         passes.add(createConstantPropagationPass());
         passes.add(createSCCPPass());
         passes.add(createAggressiveDCEPass());
