diff -up mesa-20.2.0-rc1/src/util/u_debug_stack.c.omv~ mesa-20.2.0-rc1/src/util/u_debug_stack.c
--- mesa-20.2.0-rc1/src/util/u_debug_stack.c.omv~	2020-08-09 19:46:22.109959412 +0200
+++ mesa-20.2.0-rc1/src/util/u_debug_stack.c	2020-08-09 19:46:26.785120395 +0200
@@ -94,7 +94,7 @@ debug_backtrace_capture(struct debug_sta
    unw_proc_info_t pip;
    unsigned i = 0;
 
-   pip.unwind_info = NULL;
+   pip.unwind_info = 0L;
 
    unw_getcontext(&context);
    unw_init_local(&cursor, &context);
