diff -up mesa-20.0.0-rc1/src/amd/addrlib/inc/addrtypes.h.omv~ mesa-20.0.0-rc1/src/amd/addrlib/inc/addrtypes.h
--- mesa-20.0.0-rc1/src/amd/addrlib/inc/addrtypes.h.omv~	2020-02-05 15:05:47.236576673 +0100
+++ mesa-20.0.0-rc1/src/amd/addrlib/inc/addrtypes.h	2020-02-05 15:08:09.030582925 +0100
@@ -87,7 +87,7 @@ typedef int            INT;
 #endif
 
 #ifndef ADDR_FASTCALL
-    #if defined(BRAHMA_ARM)
+    #if defined(BRAHMA_ARM) || (defined(__GNUC__) && !defined(__x86_64__) && !defined(__i386__))
         #define ADDR_FASTCALL
     #elif defined(__GNUC__)
         #define ADDR_FASTCALL __attribute__((regparm(0)))
