--- mesa-17.0.0-rc1/configure.ac.omv~	2017-01-20 18:59:53.615983644 +0100
+++ mesa-17.0.0-rc1/configure.ac	2017-01-20 19:00:29.588181260 +0100
@@ -951,6 +951,7 @@ strip_unwanted_llvm_flags() {
         -e 's/-D_GNU_SOURCE\>//g' \
         -e 's/-pedantic\>//g' \
         -e 's/-Wcovered-switch-default\>//g' \
+        -e 's/-Wstring-conversion\>//g' \
         -e 's/-O.\>//g' \
         -e 's/-g\>//g' \
         -e 's/-Wall\>//g' \
