--- Mesa-10.4.0-rc4/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp.omv~	2014-12-06 19:09:01.199748321 +0100
+++ Mesa-10.4.0-rc4/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp	2014-12-06 19:18:27.860664085 +0100
@@ -500,8 +500,12 @@ lp_build_create_jit_compiler_for_module(
        MM = new ShaderMemoryManager(JMM);
        *OutCode = MM->getGeneratedCode();
 
+#if HAVE_LLVM >= 0x0306
+       builder.setMCJITMemoryManager(std::unique_ptr<RTDyldMemoryManager>(MM));
+#else
        builder.setMCJITMemoryManager(MM);
 #endif
+#endif
    } else {
 #if HAVE_LLVM < 0x0306
        BaseMemoryManager* JMM = reinterpret_cast<BaseMemoryManager*>(CMM);
