--- mesa-17.3.0/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp.omv~	2017-12-09 01:25:31.680766004 +0100
+++ mesa-17.3.0/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp	2017-12-09 01:34:07.832980924 +0100
@@ -830,7 +830,11 @@ lp_create_builder(LLVMContextRef ctx, en
       llvm::unwrap(builder)->setFastMathFlags(flags);
       break;
    case LP_FLOAT_MODE_UNSAFE_FP_MATH:
+#if HAVE_LLVM >= 0x0600
+      flags.setFast();
+#else
       flags.setUnsafeAlgebra();
+#endif
       llvm::unwrap(builder)->setFastMathFlags(flags);
       break;
    }
