--- mesa-21.0.2/src/gallium/drivers/panfrost/pan_screen.c.org	2021-04-07 16:36:56.903008200 +0000
+++ mesa-21.0.2/src/gallium/drivers/panfrost/pan_screen.c	2021-04-10 07:06:57.849702419 +0000
@@ -115,6 +115,9 @@
         case PIPE_CAP_DEPTH_CLIP_DISABLE_SEPARATE:
         case PIPE_CAP_MIXED_COLORBUFFER_FORMATS:
         case PIPE_CAP_MIXED_FRAMEBUFFER_SIZES:
+        case PIPE_CAP_FRONTEND_NOOP:
+        case PIPE_CAP_SAMPLE_SHADING:
+        case PIPE_CAP_FRAGMENT_SHADER_DERIVATIVES:
                 return 1;
 
         case PIPE_CAP_MAX_RENDER_TARGETS:
@@ -125,10 +128,6 @@
         case PIPE_CAP_MAX_DUAL_SOURCE_RENDER_TARGETS:
                 return 1;
 
-        case PIPE_CAP_SAMPLE_SHADING:
-                /* WIP */
-                return is_gl3 ? 1 : 0;
-
         case PIPE_CAP_OCCLUSION_QUERY:
         case PIPE_CAP_PRIMITIVE_RESTART:
         case PIPE_CAP_PRIMITIVE_RESTART_FIXED_INDEX:
