--- mesa-17.3.2/src/gallium/drivers/swr/rasterizer/jitter/JitManager.cpp.omv~	2018-01-16 16:28:24.295847652 +0100
+++ mesa-17.3.2/src/gallium/drivers/swr/rasterizer/jitter/JitManager.cpp	2018-01-16 16:28:31.180825482 +0100
@@ -498,7 +498,7 @@ std::unique_ptr<llvm::MemoryBuffer> JitC
             break;
         }
 
-        pBuf = llvm::MemoryBuffer::getNewUninitMemBuffer(size_t(header.GetBufferSize()));
+        pBuf = llvm::WritableMemoryBuffer::getNewUninitMemBuffer(size_t(header.GetBufferSize()));
         if (!fread(const_cast<char*>(pBuf->getBufferStart()), header.GetBufferSize(), 1, fpIn))
         {
             pBuf = nullptr;
