diff -up mesa-19.1.0-rc2/src/glx/g_glxglvnddispatchfuncs.c.omv~ mesa-19.1.0-rc2/src/glx/g_glxglvnddispatchfuncs.c
--- mesa-19.1.0-rc2/src/glx/g_glxglvnddispatchfuncs.c.omv~	2019-05-19 15:49:37.278203648 +0200
+++ mesa-19.1.0-rc2/src/glx/g_glxglvnddispatchfuncs.c	2019-05-19 15:52:26.775930013 +0200
@@ -143,7 +143,7 @@ static GLXFBConfigSGIX *dispatch_ChooseF
     if (pChooseFBConfigSGIX == NULL)
         return NULL;
 
-    ret = (*pChooseFBConfigSGIX)(dpy, screen, attrib_list, nelements);
+    ret = (*pChooseFBConfigSGIX)(dpy, screen, (int*)attrib_list, nelements);
     if (AddFBConfigsMapping(dpy, ret, nelements, dd)) {
         free(ret);
         return NULL;
@@ -234,7 +234,7 @@ static GLXPbuffer dispatch_CreateGLXPbuf
     if (pCreateGLXPbufferSGIX == NULL)
         return None;
 
-    ret = (*pCreateGLXPbufferSGIX)(dpy, config, width, height, attrib_list);
+    ret = (*pCreateGLXPbufferSGIX)(dpy, config, width, height, (int*)attrib_list);
     if (AddDrawableMapping(dpy, ret, dd)) {
         PFNGLXDESTROYGLXPBUFFERSGIXPROC pDestroyGLXPbufferSGIX;
 
diff -up mesa-19.1.0-rc2/src/glx/glxglvnd.c.omv~ mesa-19.1.0-rc2/src/glx/glxglvnd.c
--- mesa-19.1.0-rc2/src/glx/glxglvnd.c.omv~	2019-05-19 15:53:36.667481737 +0200
+++ mesa-19.1.0-rc2/src/glx/glxglvnd.c	2019-05-19 15:53:43.468439825 +0200
@@ -41,7 +41,7 @@ static void *__glXGLVNDGetDispatchAddres
 {
     unsigned internalIndex = FindGLXFunction(procName);
 
-    return __glXDispatchFunctions[internalIndex];
+    return (void*)__glXDispatchFunctions[internalIndex];
 }
 
 static void __glXGLVNDSetDispatchIndex(const GLubyte *procName, int index)
