diff -up mesa-19.2.0/src/gallium/state_trackers/xvmc/meson.build.omv~ mesa-19.2.0/src/gallium/state_trackers/xvmc/meson.build
--- mesa-19.2.0/src/gallium/state_trackers/xvmc/meson.build.omv~	2019-09-27 15:57:08.778929637 +0200
+++ mesa-19.2.0/src/gallium/state_trackers/xvmc/meson.build	2019-09-27 16:03:06.224667267 +0200
@@ -23,7 +23,7 @@ libxvmc_st = static_library(
   files('attributes.c', 'block.c', 'context.c', 'surface.c', 'subpicture.c'),
   c_args : [c_vis_args],
   include_directories : [inc_common],
-  dependencies : [dep_xvmc, dep_x11_xcb, dep_xcb, dep_xcb_dri2, dep_xcb_dri3],
+  dependencies : [dep_xvmc, dep_x11_xcb, dep_xcb, dep_xcb_dri2, dep_xcb_dri3, cc.find_library('Xv')],
 )
 
 # These tests will not work without a working xvmc configuration. 
@@ -36,7 +36,7 @@ if with_tools.contains('xvmc')
       _name,
       files('tests/test_@0@.c'.format(x), 'tests/testlib.c'),
       dependencies : [
-        dep_real_xvmc, dep_x11_xcb, dep_xcb, dep_xcb_dri2, dep_xcb_dri3, dep_xvmcw,
+        dep_real_xvmc, dep_x11_xcb, dep_xcb, dep_xcb_dri2, dep_xcb_dri3, dep_xvmcw, cc.find_library('Xv')
       ],
       install: true,
     )
@@ -46,7 +46,7 @@ if with_tools.contains('xvmc')
     'xvmc_bench',
     files('tests/xvmc_bench.c', 'tests/testlib.c'),
     dependencies : [
-      dep_real_xvmc, dep_x11_xcb, dep_xcb, dep_xcb_dri2, dep_xcb_dri3, dep_xvmcw,
+      dep_real_xvmc, dep_x11_xcb, dep_xcb, dep_xcb_dri2, dep_xcb_dri3, dep_xvmcw, cc.find_library('Xv')
     ],
     install: true,
   )
