--- mesa-13.0-20160910/src/amd/addrlib/addrtypes.h.omv~	2016-09-10 09:57:02.666266691 +0200
+++ mesa-13.0-20160910/src/amd/addrlib/addrtypes.h	2016-09-10 09:57:37.353727052 +0200
@@ -87,7 +87,7 @@ typedef int            INT;
 #endif
 
 #ifndef ADDR_FASTCALL
-    #if defined(__GNUC__)
+    #if defined(__GNUC__) && (defined(__i386__) || defined(__x86_64__))
         #define ADDR_FASTCALL __attribute__((regparm(0)))
     #else
         #define ADDR_FASTCALL __fastcall
--- mesa-13.0-20160910/src/amd/addrlib/inc/lnx_common_defs.h.omv~	2016-09-10 09:57:47.627566064 +0200
+++ mesa-13.0-20160910/src/amd/addrlib/inc/lnx_common_defs.h	2016-09-10 09:58:07.493253340 +0200
@@ -46,7 +46,11 @@ typedef int           __kernel_ptrdiff_t
 
 
 #if !defined(ATI_API_CALL)
+#if defined(__i386__) || defined(__x86_64__)
 #define ATI_API_CALL __attribute__((regparm(0)))
+#else
+#define ATI_API_CALL
+#endif
 #endif
 
 extern void * ATI_API_CALL __ke_memset(void* s, int c, __ke_size_t count);
