diff -up mesa-22.3.0-rc2/meson.build.3~ mesa-22.3.0-rc2/meson.build
--- mesa-22.3.0-rc2/meson.build.3~	2022-11-10 13:32:08.447637431 +0100
+++ mesa-22.3.0-rc2/meson.build	2022-11-10 13:36:38.602764575 +0100
@@ -182,6 +182,12 @@ endif
 
 gallium_drivers = get_option('gallium-drivers')
 if gallium_drivers.contains('auto')
+  extras = []
+  foreach drv : gallium_drivers
+    if drv != 'auto'
+      extras += drv
+    endif
+  endforeach
   if system_has_kms_drm
     # TODO: PPC, Sparc
     if ['x86', 'x86_64'].contains(host_machine.cpu_family())
@@ -209,6 +215,11 @@ if gallium_drivers.contains('auto')
     error('Unknown OS @0@. Please pass -Dgallium-drivers to set driver options. Patches gladly accepted to fix this.'.format(
           host_machine.system()))
   endif
+  foreach drv : extras
+    if not gallium_drivers.contains(drv)
+      gallium_drivers += drv
+    endif
+  endforeach
 endif
 with_gallium_radeonsi = gallium_drivers.contains('radeonsi')
 with_gallium_r300 = gallium_drivers.contains('r300')
@@ -252,6 +263,12 @@ endif
 
 _vulkan_drivers = get_option('vulkan-drivers')
 if _vulkan_drivers.contains('auto')
+  extras = []
+  foreach drv : _vulkan_drivers
+    if drv != 'auto'
+      extras += drv
+    endif
+  endforeach
   if system_has_kms_drm
     if host_machine.cpu_family().startswith('x86')
       _vulkan_drivers = ['amd', 'intel', 'intel_hasvk', 'swrast']
@@ -268,6 +285,11 @@ if _vulkan_drivers.contains('auto')
     error('Unknown OS @0@. Please pass -Dvulkan-drivers to set driver options. Patches gladly accepted to fix this.'.format(
           host_machine.system()))
   endif
+  foreach drv : extras
+    if not _vulkan_drivers.contains(drv)
+      _vulkan_drivers += drv
+    endif
+  endforeach
 endif
 
 with_intel_vk = _vulkan_drivers.contains('intel')
