--- mesa-13.0-20161014/src/util/bitscan.h.omv~	2016-10-14 15:52:57.095208728 +0200
+++ mesa-13.0-20161014/src/util/bitscan.h	2016-10-14 15:56:52.296879490 +0200
@@ -48,6 +48,7 @@ extern "C" {
  * Return 0 if no bits set.
  */
 #ifdef HAVE___BUILTIN_FFS
+#include <string.h> /* Get libc's definition of ffs out of the way */
 #define ffs __builtin_ffs
 #elif defined(_MSC_VER) && (_M_IX86 || _M_ARM || _M_AMD64 || _M_IA64)
 static inline
@@ -60,8 +61,7 @@ int ffs(unsigned i)
       return 0;
 }
 #else
-extern
-int ffs(unsigned i);
+#include <string.h>
 #endif
 
 #ifdef HAVE___BUILTIN_FFSLL
