diff -up mesa-19.1.0-rc2/meson.build.2~ mesa-19.1.0-rc2/meson.build
--- mesa-19.1.0-rc2/meson.build.2~	2019-05-14 15:43:46.000000000 +0200
+++ mesa-19.1.0-rc2/meson.build	2019-05-19 15:44:23.247727825 +0200
@@ -116,7 +116,7 @@ if dri_drivers.contains('auto')
     if ['x86', 'x86_64'].contains(host_machine.cpu_family())
       dri_drivers = ['i915', 'i965', 'r100', 'r200', 'nouveau']
     elif ['arm', 'aarch64'].contains(host_machine.cpu_family())
-      dri_drivers = []
+      dri_drivers = ['r100', 'r200', 'nouveau']
     else
       error('Unknown architecture @0@. Please pass -Ddri-drivers to set driver options. Patches gladly accepted to fix this.'.format(
             host_machine.cpu_family()))
@@ -149,8 +149,9 @@ if gallium_drivers.contains('auto')
       ]
     elif ['arm', 'aarch64'].contains(host_machine.cpu_family())
       gallium_drivers = [
-        'kmsro', 'v3d', 'vc4', 'freedreno', 'etnaviv', 'nouveau',
-        'tegra', 'virgl', 'lima', 'swrast'
+        'kmsro', 'v3d', 'vc4', 'freedreno', 'etnaviv', 'r300',
+        'r600', 'radeonsi', 'nouveau',
+        'tegra', 'panfrost', 'virgl', 'lima', 'swrast'
       ]
     else
       error('Unknown architecture @0@. Please pass -Dgallium-drivers to set driver options. Patches gladly accepted to fix this.'.format(
@@ -206,7 +207,7 @@ if _vulkan_drivers.contains('auto')
     if host_machine.cpu_family().startswith('x86')
       _vulkan_drivers = ['amd', 'intel']
     elif ['arm', 'aarch64'].contains(host_machine.cpu_family())
-      _vulkan_drivers = []
+      _vulkan_drivers = ['amd']
     else
       error('Unknown architecture @0@. Please pass -Dvulkan-drivers to set driver options. Patches gladly accepted to fix this.'.format(
             host_machine.cpu_family()))
