--- mesa-10.6.0-rc2/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp.omv~	2015-05-27 20:50:32.274020038 +0200
+++ mesa-10.6.0-rc2/src/gallium/auxiliary/gallivm/lp_bld_misc.cpp	2015-05-27 21:01:06.065238353 +0200
@@ -433,15 +433,6 @@ lp_build_create_jit_compiler_for_module(
    options.JITEmitDebugInfo = true;
 #endif
 
-   /* XXX: Workaround http://llvm.org/PR21435 */
-#if defined(DEBUG) || defined(PROFILE) || \
-    (HAVE_LLVM >= 0x0303 && (defined(PIPE_ARCH_X86) || defined(PIPE_ARCH_X86_64)))
-#if HAVE_LLVM < 0x0304
-   options.NoFramePointerElimNonLeaf = true;
-#endif
-   options.NoFramePointerElim = true;
-#endif
-
    builder.setEngineKind(EngineKind::JIT)
           .setErrorStr(&Error)
           .setTargetOptions(options)
