diff -up mesa-18.2.0-rc1/src/gallium/drivers/swr/rasterizer/jitter/gen_builder.hpp.omv~ mesa-18.2.0-rc1/src/gallium/drivers/swr/rasterizer/jitter/gen_builder.hpp
--- mesa-18.2.0-rc1/src/gallium/drivers/swr/rasterizer/jitter/gen_builder.hpp.omv~	2018-08-05 02:25:26.740457313 +0200
+++ mesa-18.2.0-rc1/src/gallium/drivers/swr/rasterizer/jitter/gen_builder.hpp	2018-08-05 02:32:22.156038196 +0200
@@ -58,32 +58,32 @@ CallInst* MEMSET(Value *Ptr, Value *Val,
 
 CallInst* MEMCOPY(Value *Dst, Value *Src, uint64_t Size, unsigned Align, bool isVolatile = false, MDNode *TBAATag = nullptr, MDNode *TBAAStructTag = nullptr, MDNode *ScopeTag = nullptr, MDNode *NoAliasTag = nullptr)
 {
-    return IRB()->CreateMemCpy(Dst, Src, Size, Align, isVolatile, TBAATag, TBAAStructTag, ScopeTag, NoAliasTag);
+    return IRB()->CreateMemCpy(Dst, Align, Src, Align, Size, isVolatile, TBAATag, TBAAStructTag, ScopeTag, NoAliasTag);
 }
 
 CallInst* MEMCOPY(Value *Dst, Value *Src, Value *Size, unsigned Align, bool isVolatile = false, MDNode *TBAATag = nullptr, MDNode *TBAAStructTag = nullptr, MDNode *ScopeTag = nullptr, MDNode *NoAliasTag = nullptr)
 {
-    return IRB()->CreateMemCpy(Dst, Src, Size, Align, isVolatile, TBAATag, TBAAStructTag, ScopeTag, NoAliasTag);
+    return IRB()->CreateMemCpy(Dst, Align, Src, Align, Size, isVolatile, TBAATag, TBAAStructTag, ScopeTag, NoAliasTag);
 }
 
 CallInst* ELEMENT_UNORDERED_ATOMIC_MEM_CPY(Value *Dst, Value *Src, uint64_t Size, uint32_t ElementSize, MDNode *TBAATag = nullptr, MDNode *TBAAStructTag = nullptr, MDNode *ScopeTag = nullptr, MDNode *NoAliasTag = nullptr)
 {
-    return IRB()->CreateElementUnorderedAtomicMemCpy(Dst, Src, Size, ElementSize, TBAATag, TBAAStructTag, ScopeTag, NoAliasTag);
+    return IRB()->CreateElementUnorderedAtomicMemCpy(Dst, 0, Src, 0, Size, ElementSize, TBAATag, TBAAStructTag, ScopeTag, NoAliasTag);
 }
 
 CallInst* ELEMENT_UNORDERED_ATOMIC_MEM_CPY(Value *Dst, Value *Src, Value *Size, uint32_t ElementSize, MDNode *TBAATag = nullptr, MDNode *TBAAStructTag = nullptr, MDNode *ScopeTag = nullptr, MDNode *NoAliasTag = nullptr)
 {
-    return IRB()->CreateElementUnorderedAtomicMemCpy(Dst, Src, Size, ElementSize, TBAATag, TBAAStructTag, ScopeTag, NoAliasTag);
+    return IRB()->CreateElementUnorderedAtomicMemCpy(Dst, 0, Src, 0, Size, ElementSize, TBAATag, TBAAStructTag, ScopeTag, NoAliasTag);
 }
 
 CallInst* MEMMOVE(Value *Dst, Value *Src, uint64_t Size, unsigned Align, bool isVolatile = false, MDNode *TBAATag = nullptr, MDNode *ScopeTag = nullptr, MDNode *NoAliasTag = nullptr)
 {
-    return IRB()->CreateMemMove(Dst, Src, Size, Align, isVolatile, TBAATag, ScopeTag, NoAliasTag);
+    return IRB()->CreateMemMove(Dst, Align, Src, Align, Size, isVolatile, TBAATag, ScopeTag, NoAliasTag);
 }
 
 CallInst* MEMMOVE(Value *Dst, Value *Src, Value *Size, unsigned Align, bool isVolatile = false, MDNode *TBAATag = nullptr, MDNode *ScopeTag = nullptr, MDNode *NoAliasTag = nullptr)
 {
-    return IRB()->CreateMemMove(Dst, Src, Size, Align, isVolatile, TBAATag, ScopeTag, NoAliasTag);
+    return IRB()->CreateMemMove(Dst, Align, Src, Align, Size, isVolatile, TBAATag, ScopeTag, NoAliasTag);
 }
 
 CallInst* FADD_REDUCE(Value *Acc, Value *Src)
@@ -988,7 +988,7 @@ Value* PTR_DIFF(Value *LHS, Value *RHS,
 
 Value* INVARIANT_GROUP_BARRIER(Value *Ptr)
 {
-    return IRB()->CreateInvariantGroupBarrier(Ptr);
+    return IRB()->CreateLaunderInvariantGroup(Ptr);
 }
 
 Value* VECTOR_SPLAT(unsigned NumElts, Value *V, const Twine &Name = "")
