diff -up mesa-20.1.1/src/gallium/frontends/clover/api/clApiRename.h.omv~ mesa-20.1.1/src/gallium/frontends/clover/api/clApiRename.h
--- mesa-20.1.1/src/gallium/frontends/clover/api/clApiRename.h.omv~	2020-06-13 04:13:36.516201275 +0200
+++ mesa-20.1.1/src/gallium/frontends/clover/api/clApiRename.h	2020-06-13 04:13:29.995130298 +0200
@@ -0,0 +1,148 @@
+#pragma once
+
+#define clGetPlatformIDs clGetPlatformIDsMESA
+#define GetPlatformInfo GetPlatformInfoMESA
+#define clGetDeviceIDs clGetDeviceIDsMESA
+#define clGetDeviceInfo clGetDeviceInfoMESA
+#define clCreateContext clCreateContextMESA
+#define clCreateContextFromType clCreateContextFromTypeMESA
+#define clRetainContext clRetainContextMESA
+#define clReleaseContext clReleaseContextMESA
+#define clGetContextInfo clGetContextInfoMESA
+#define clCreateCommandQueue clCreateCommandQueueMESA
+#define clRetainCommandQueue clRetainCommandQueueMESA
+#define clReleaseCommandQueue clReleaseCommandQueueMESA
+#define clGetCommandQueueInfo clGetCommandQueueInfoMESA
+#define clSetCommandQueueProperty clSetCommandQueuePropertyMESA
+#define clCreateBuffer clCreateBufferMESA
+#define clCreateImage2D clCreateImage2DMESA
+#define clCreateImage3D clCreateImage3DMESA
+#define clRetainMemObject clRetainMemObjectMESA
+#define clReleaseMemObject clReleaseMemObjectMESA
+#define clGetSupportedImageFormats clGetSupportedImageFormatsMESA
+#define clGetMemObjectInfo clGetMemObjectInfoMESA
+#define clGetImageInfo clGetImageInfoMESA
+#define clCreateSampler clCreateSamplerMESA
+#define clRetainSampler clRetainSamplerMESA
+#define clReleaseSampler clReleaseSamplerMESA
+#define clGetSamplerInfo clGetSamplerInfoMESA
+#define clCreateProgramWithSource clCreateProgramWithSourceMESA
+#define clCreateProgramWithBinary clCreateProgramWithBinaryMESA
+#define clRetainProgram clRetainProgramMESA
+#define clReleaseProgram clReleaseProgramMESA
+#define clBuildProgram clBuildProgramMESA
+#define clUnloadCompiler clUnloadCompilerMESA
+#define clGetProgramInfo clGetProgramInfoMESA
+#define clGetProgramBuildInfo clGetProgramBuildInfoMESA
+#define clCreateKernel clCreateKernelMESA
+#define clCreateKernelsInProgram clCreateKernelsInProgramMESA
+#define clRetainKernel clRetainKernelMESA
+#define clReleaseKernel clReleaseKernelMESA
+#define clSetKernelArg clSetKernelArgMESA
+#define clGetKernelInfo clGetKernelInfoMESA
+#define clGetKernelWorkGroupInfo clGetKernelWorkGroupInfoMESA
+#define clWaitForEvents clWaitForEventsMESA
+#define clGetEventInfo clGetEventInfoMESA
+#define clRetainEvent clRetainEventMESA
+#define clReleaseEvent clReleaseEventMESA
+#define clGetEventProfilingInfo clGetEventProfilingInfoMESA
+#define clFlush clFlushMESA
+#define clFinish clFinishMESA
+#define clEnqueueReadBuffer clEnqueueReadBufferMESA
+#define clEnqueueWriteBuffer clEnqueueWriteBufferMESA
+#define clEnqueueCopyBuffer clEnqueueCopyBufferMESA
+#define clEnqueueReadImage clEnqueueReadImageMESA
+#define clEnqueueWriteImage clEnqueueWriteImageMESA
+#define clEnqueueCopyImage clEnqueueCopyImageMESA
+#define clEnqueueCopyImageToBuffer clEnqueueCopyImageToBufferMESA
+#define clEnqueueCopyBufferToImage clEnqueueCopyBufferToImageMESA
+#define clEnqueueMapBuffer clEnqueueMapBufferMESA
+#define clEnqueueMapImage clEnqueueMapImageMESA
+#define clEnqueueUnmapMemObject clEnqueueUnmapMemObjectMESA
+#define clEnqueueNDRangeKernel clEnqueueNDRangeKernelMESA
+#define clEnqueueTask clEnqueueTaskMESA
+#define clEnqueueNativeKernel clEnqueueNativeKernelMESA
+#define clEnqueueMarker clEnqueueMarkerMESA
+#define clEnqueueWaitForEvents clEnqueueWaitForEventsMESA
+#define clEnqueueBarrier clEnqueueBarrierMESA
+#define GetExtensionFunctionAddress GetExtensionFunctionAddressMESA
+#define clCreateFromGLBuffer clCreateFromGLBufferMESA
+#define gclCreateFromGLTexture2D gclCreateFromGLTexture2DMESA
+#define gclCreateFromGLTexture3D gclCreateFromGLTexture3DMESA
+#define gclCreateFromGLRenderbuffer gclCreateFromGLRenderbufferMESA
+#define gclGetGLObjectInfo gclGetGLObjectInfoMESA
+#define gclGetGLTextureInfo gclGetGLTextureInfoMESA
+#define gclEnqueueAcquireGLObjects gclEnqueueAcquireGLObjectsMESA
+#define gclEnqueueReleaseGLObjects gclEnqueueReleaseGLObjectsMESA
+#define gclGetGLContextInfoKHR gclGetGLContextInfoKHRMESA
+#define gclGetDeviceIDsFromD3D10KHR gclGetDeviceIDsFromD3D10KHRMESA
+#define gclCreateFromD3D10BufferKHR gclCreateFromD3D10BufferKHRMESA
+#define gclCreateFromD3D10Texture2DKHR gclCreateFromD3D10Texture2DKHRMESA
+#define gclCreateFromD3D10Texture3DKHR gclCreateFromD3D10Texture3DKHRMESA
+#define gclEnqueueAcquireD3D10ObjectsKHR gclEnqueueAcquireD3D10ObjectsKHRMESA
+#define gclEnqueueReleaseD3D10ObjectsKHR gclEnqueueReleaseD3D10ObjectsKHRMESA
+#define clSetEventCallback clSetEventCallbackMESA
+#define clCreateSubBuffer clCreateSubBufferMESA
+#define clSetMemObjectDestructorCallback clSetMemObjectDestructorCallbackMESA
+#define clCreateUserEvent clCreateUserEventMESA
+#define clSetUserEventStatus clSetUserEventStatusMESA
+#define clEnqueueReadBufferRect clEnqueueReadBufferRectMESA
+#define clEnqueueWriteBufferRect clEnqueueWriteBufferRectMESA
+#define clEnqueueCopyBufferRect clEnqueueCopyBufferRectMESA
+#define gclCreateSubDevicesEXT gclCreateSubDevicesEXTMESA
+#define gclRetainDeviceEXT gclRetainDeviceEXTMESA
+#define gclReleaseDeviceEXT gclReleaseDeviceEXTMESA
+#define gclCreateEventFromGLsyncKHR gclCreateEventFromGLsyncKHRMESA
+#define clCreateSubDevices clCreateSubDevicesMESA
+#define clRetainDevice clRetainDeviceMESA
+#define clReleaseDevice clReleaseDeviceMESA
+#define clCreateImage clCreateImageMESA
+#define clCreateProgramWithBuiltInKernels clCreateProgramWithBuiltInKernelsMESA
+#define clCompileProgram clCompileProgramMESA
+#define clLinkProgram clLinkProgramMESA
+#define clUnloadPlatformCompiler clUnloadPlatformCompilerMESA
+#define clGetKernelArgInfo clGetKernelArgInfoMESA
+#define clEnqueueFillBuffer clEnqueueFillBufferMESA
+#define clEnqueueFillImage clEnqueueFillImageMESA
+#define clEnqueueMigrateMemObjects clEnqueueMigrateMemObjectsMESA
+#define clEnqueueMarkerWithWaitList clEnqueueMarkerWithWaitListMESA
+#define clEnqueueBarrierWithWaitList clEnqueueBarrierWithWaitListMESA
+#define GetExtensionFunctionAddressForPlatform GetExtensionFunctionAddressForPlatformMESA
+#define gclCreateFromGLTexture gclCreateFromGLTextureMESA
+#define gclGetDeviceIDsFromD3D11KHR gclGetDeviceIDsFromD3D11KHRMESA
+#define gclCreateFromD3D11BufferKHR gclCreateFromD3D11BufferKHRMESA
+#define gclCreateFromD3D11Texture2DKHR gclCreateFromD3D11Texture2DKHRMESA
+#define gclCreateFromD3D11Texture3DKHR gclCreateFromD3D11Texture3DKHRMESA
+#define gclCreateFromDX9MediaSurfaceKHR gclCreateFromDX9MediaSurfaceKHRMESA
+#define gclEnqueueAcquireD3D11ObjectsKHR gclEnqueueAcquireD3D11ObjectsKHRMESA
+#define gclEnqueueReleaseD3D11ObjectsKHR gclEnqueueReleaseD3D11ObjectsKHRMESA
+#define gclGetDeviceIDsFromDX9MediaAdapterKHR gclGetDeviceIDsFromDX9MediaAdapterKHRMESA
+#define gclEnqueueAcquireDX9MediaSurfacesKHR gclEnqueueAcquireDX9MediaSurfacesKHRMESA
+#define gclEnqueueReleaseDX9MediaSurfacesKHR gclEnqueueReleaseDX9MediaSurfacesKHRMESA
+#define gclCreateFromEGLImageKHR gclCreateFromEGLImageKHRMESA
+#define gclEnqueueAcquireEGLObjectsKHR gclEnqueueAcquireEGLObjectsKHRMESA
+#define gclEnqueueReleaseEGLObjectsKHR gclEnqueueReleaseEGLObjectsKHRMESA
+#define gclCreateEventFromEGLSyncKHR gclCreateEventFromEGLSyncKHRMESA
+#define clCreateCommandQueueWithProperties clCreateCommandQueueWithPropertiesMESA
+#define gclCreatePipe gclCreatePipeMESA
+#define gclGetPipeInfo gclGetPipeInfoMESA
+#define clSVMAlloc clSVMAllocMESA
+#define clSVMFree clSVMFreeMESA
+#define clEnqueueSVMFree clEnqueueSVMFreeMESA
+#define clEnqueueSVMMemcpy clEnqueueSVMMemcpyMESA
+#define clEnqueueSVMMemFill clEnqueueSVMMemFillMESA
+#define clEnqueueSVMMap clEnqueueSVMMapMESA
+#define clEnqueueSVMUnmap clEnqueueSVMUnmapMESA
+#define gclCreateSamplerWithProperties gclCreateSamplerWithPropertiesMESA
+#define clSetKernelArgSVMPointer clSetKernelArgSVMPointerMESA
+#define clSetKernelExecInfo clSetKernelExecInfoMESA
+#define gclGetKernelSubGroupInfoKHR gclGetKernelSubGroupInfoKHRMESA
+#define gclCloneKernel gclCloneKernelMESA
+#define gclCreateProgramWithIL gclCreateProgramWithILMESA
+#define clEnqueueSVMMigrateMem clEnqueueSVMMigrateMemMESA
+#define gclGetDeviceAndHostTimer gclGetDeviceAndHostTimerMESA
+#define gclGetHostTimer gclGetHostTimerMESA
+#define gclGetKernelSubGroupInfo gclGetKernelSubGroupInfoMESA
+#define gclSetDefaultDeviceCommandQueue gclSetDefaultDeviceCommandQueueMESA
+#define gclSetProgramReleaseCallback gclSetProgramReleaseCallbackMESA
+#define gclSetProgramSpecializationConstant gclSetProgramSpecializationConstantMESA
diff -up mesa-20.1.1/src/gallium/frontends/clover/api/context.cpp.omv~ mesa-20.1.1/src/gallium/frontends/clover/api/context.cpp
--- mesa-20.1.1/src/gallium/frontends/clover/api/context.cpp.omv~	2020-06-13 04:13:54.129393001 +0200
+++ mesa-20.1.1/src/gallium/frontends/clover/api/context.cpp	2020-06-13 04:14:02.789490165 +0200
@@ -20,6 +20,7 @@
 // OTHER DEALINGS IN THE SOFTWARE.
 //
 
+#include "clApiRename.h"
 #include "api/util.hpp"
 #include "core/context.hpp"
 #include "core/platform.hpp"
diff -up mesa-20.1.1/src/gallium/frontends/clover/api/device.cpp.omv~ mesa-20.1.1/src/gallium/frontends/clover/api/device.cpp
--- mesa-20.1.1/src/gallium/frontends/clover/api/device.cpp.omv~	2020-06-13 04:14:11.436593163 +0200
+++ mesa-20.1.1/src/gallium/frontends/clover/api/device.cpp	2020-06-13 04:14:29.429807081 +0200
@@ -20,6 +20,7 @@
 // OTHER DEALINGS IN THE SOFTWARE.
 //
 
+#include "clApiRename.h"
 #include "api/util.hpp"
 #include "core/platform.hpp"
 #include "core/device.hpp"
diff -up mesa-20.1.1/src/gallium/frontends/clover/api/dispatch.cpp.omv~ mesa-20.1.1/src/gallium/frontends/clover/api/dispatch.cpp
--- mesa-20.1.1/src/gallium/frontends/clover/api/dispatch.cpp.omv~	2020-06-13 04:09:49.090728678 +0200
+++ mesa-20.1.1/src/gallium/frontends/clover/api/dispatch.cpp	2020-06-13 04:14:48.862037518 +0200
@@ -20,6 +20,7 @@
 // OTHER DEALINGS IN THE SOFTWARE.
 //
 
+#include "clApiRename.h"
 #include "api/dispatch.hpp"
 
 namespace clover {
diff -up mesa-20.1.1/src/gallium/frontends/clover/api/dispatch.hpp.omv~ mesa-20.1.1/src/gallium/frontends/clover/api/dispatch.hpp
--- mesa-20.1.1/src/gallium/frontends/clover/api/dispatch.hpp.omv~	2020-06-13 04:14:54.499104254 +0200
+++ mesa-20.1.1/src/gallium/frontends/clover/api/dispatch.hpp	2020-06-13 04:15:06.164242198 +0200
@@ -23,6 +23,7 @@
 #ifndef API_DISPATCH_HPP
 #define API_DISPATCH_HPP
 
+#include "clApiRename.h"
 #include "CL/cl.h"
 #include "CL/cl_ext.h"
 #include "CL/cl_egl.h"
diff -up mesa-20.1.1/src/gallium/frontends/clover/api/event.cpp.omv~ mesa-20.1.1/src/gallium/frontends/clover/api/event.cpp
--- mesa-20.1.1/src/gallium/frontends/clover/api/event.cpp.omv~	2020-06-13 04:15:13.022323201 +0200
+++ mesa-20.1.1/src/gallium/frontends/clover/api/event.cpp	2020-06-13 04:15:21.093418441 +0200
@@ -20,6 +20,7 @@
 // OTHER DEALINGS IN THE SOFTWARE.
 //
 
+#include "clApiRename.h"
 #include "api/util.hpp"
 #include "core/event.hpp"
 
diff -up mesa-20.1.1/src/gallium/frontends/clover/api/interop.cpp.omv~ mesa-20.1.1/src/gallium/frontends/clover/api/interop.cpp
--- mesa-20.1.1/src/gallium/frontends/clover/api/interop.cpp.omv~	2020-06-13 04:15:24.803462188 +0200
+++ mesa-20.1.1/src/gallium/frontends/clover/api/interop.cpp	2020-06-13 04:15:32.979558526 +0200
@@ -21,6 +21,7 @@
 // OTHER DEALINGS IN THE SOFTWARE.
 //
 
+#include "clApiRename.h"
 #include "core/event.hpp"
 #include "api/util.hpp"
 
diff -up mesa-20.1.1/src/gallium/frontends/clover/api/kernel.cpp.omv~ mesa-20.1.1/src/gallium/frontends/clover/api/kernel.cpp
--- mesa-20.1.1/src/gallium/frontends/clover/api/kernel.cpp.omv~	2020-06-13 04:15:35.759591261 +0200
+++ mesa-20.1.1/src/gallium/frontends/clover/api/kernel.cpp	2020-06-13 04:15:43.675684411 +0200
@@ -20,6 +20,7 @@
 // OTHER DEALINGS IN THE SOFTWARE.
 //
 
+#include "clApiRename.h"
 #include "api/util.hpp"
 #include "core/kernel.hpp"
 #include "core/event.hpp"
diff -up mesa-20.1.1/src/gallium/frontends/clover/api/memory.cpp.omv~ mesa-20.1.1/src/gallium/frontends/clover/api/memory.cpp
--- mesa-20.1.1/src/gallium/frontends/clover/api/memory.cpp.omv~	2020-06-13 04:15:46.620719044 +0200
+++ mesa-20.1.1/src/gallium/frontends/clover/api/memory.cpp	2020-06-13 04:15:54.654813460 +0200
@@ -20,6 +20,7 @@
 // OTHER DEALINGS IN THE SOFTWARE.
 //
 
+#include "clApiRename.h"
 #include "util/u_math.h"
 #include "api/util.hpp"
 #include "core/memory.hpp"
diff -up mesa-20.1.1/src/gallium/frontends/clover/api/platform.cpp.omv~ mesa-20.1.1/src/gallium/frontends/clover/api/platform.cpp
--- mesa-20.1.1/src/gallium/frontends/clover/api/platform.cpp.omv~	2020-06-13 04:15:57.941852065 +0200
+++ mesa-20.1.1/src/gallium/frontends/clover/api/platform.cpp	2020-06-13 04:16:09.473987383 +0200
@@ -22,6 +22,7 @@
 
 #include <unordered_map>
 
+#include "clApiRename.h"
 #include "api/dispatch.hpp"
 #include "api/util.hpp"
 #include "core/platform.hpp"
diff -up mesa-20.1.1/src/gallium/frontends/clover/api/program.cpp.omv~ mesa-20.1.1/src/gallium/frontends/clover/api/program.cpp
--- mesa-20.1.1/src/gallium/frontends/clover/api/program.cpp.omv~	2020-06-13 04:16:12.324020798 +0200
+++ mesa-20.1.1/src/gallium/frontends/clover/api/program.cpp	2020-06-13 04:16:20.080111680 +0200
@@ -20,6 +20,7 @@
 // OTHER DEALINGS IN THE SOFTWARE.
 //
 
+#include "clApiRename.h"
 #include "api/util.hpp"
 #include "core/program.hpp"
 #include "util/u_debug.h"
diff -up mesa-20.1.1/src/gallium/frontends/clover/api/queue.cpp.omv~ mesa-20.1.1/src/gallium/frontends/clover/api/queue.cpp
--- mesa-20.1.1/src/gallium/frontends/clover/api/queue.cpp.omv~	2020-06-13 04:16:22.944145219 +0200
+++ mesa-20.1.1/src/gallium/frontends/clover/api/queue.cpp	2020-06-13 04:16:30.586234659 +0200
@@ -20,6 +20,7 @@
 // OTHER DEALINGS IN THE SOFTWARE.
 //
 
+#include "clApiRename.h"
 #include "api/util.hpp"
 #include "core/queue.hpp"
 
diff -up mesa-20.1.1/src/gallium/frontends/clover/api/sampler.cpp.omv~ mesa-20.1.1/src/gallium/frontends/clover/api/sampler.cpp
--- mesa-20.1.1/src/gallium/frontends/clover/api/sampler.cpp.omv~	2020-06-13 04:16:34.732283151 +0200
+++ mesa-20.1.1/src/gallium/frontends/clover/api/sampler.cpp	2020-06-13 04:16:42.802377477 +0200
@@ -20,6 +20,7 @@
 // OTHER DEALINGS IN THE SOFTWARE.
 //
 
+#include "clApiRename.h"
 #include "api/util.hpp"
 #include "core/sampler.hpp"
 
diff -up mesa-20.1.1/src/gallium/frontends/clover/api/transfer.cpp.omv~ mesa-20.1.1/src/gallium/frontends/clover/api/transfer.cpp
--- mesa-20.1.1/src/gallium/frontends/clover/api/transfer.cpp.omv~	2020-06-13 04:16:46.146416539 +0200
+++ mesa-20.1.1/src/gallium/frontends/clover/api/transfer.cpp	2020-06-13 04:16:54.231510924 +0200
@@ -22,6 +22,7 @@
 
 #include <cstring>
 
+#include "clApiRename.h"
 #include "util/bitscan.h"
 
 #include "api/dispatch.hpp"
diff -up mesa-20.1.1/src/gallium/frontends/clover/api/util.hpp.omv~ mesa-20.1.1/src/gallium/frontends/clover/api/util.hpp
--- mesa-20.1.1/src/gallium/frontends/clover/api/util.hpp.omv~	2020-06-13 04:16:56.766540502 +0200
+++ mesa-20.1.1/src/gallium/frontends/clover/api/util.hpp	2020-06-13 04:17:05.181638629 +0200
@@ -26,6 +26,7 @@
 #include <cassert>
 #include <iostream>
 
+#include "clApiRename.h"
 #include "core/error.hpp"
 #include "core/property.hpp"
 #include "util/algorithm.hpp"
