--- Mesa-9.0/src/mesa/drivers/dri/i915/intel_screen.c.bero	2012-10-09 07:34:59.585769202 +0200
+++ Mesa-9.0/src/mesa/drivers/dri/i915/intel_screen.c	2012-10-09 07:37:05.496776380 +0200
@@ -26,6 +26,7 @@
  **************************************************************************/
 
 #include <errno.h>
+#include <time.h>
 #include "main/glheader.h"
 #include "main/context.h"
 #include "main/framebuffer.h"
--- Mesa-9.0/src/mesa/drivers/dri/i965/intel_screen.c.bero	2012-10-09 07:45:52.715806434 +0200
+++ Mesa-9.0/src/mesa/drivers/dri/i965/intel_screen.c	2012-10-09 07:46:00.426806874 +0200
@@ -26,6 +26,7 @@
  **************************************************************************/
 
 #include <errno.h>
+#include <time.h>
 #include "main/glheader.h"
 #include "main/context.h"
 #include "main/framebuffer.h"
