diff -up mesa-23.2.0/src/gallium/drivers/panfrost/pan_cmdstream.c.omv~ mesa-23.2.0/src/gallium/drivers/panfrost/pan_cmdstream.c
--- mesa-23.2.0/src/gallium/drivers/panfrost/pan_cmdstream.c.omv~	2023-07-29 15:22:49.688831519 +0100
+++ mesa-23.2.0/src/gallium/drivers/panfrost/pan_cmdstream.c	2023-07-29 15:22:58.758854994 +0100
@@ -3101,8 +3101,8 @@ pan_allow_rotating_primitives(const stru
                               const struct pipe_draw_info *info)
 {
    bool lines =
-      (info->mode == PIPE_PRIM_LINES || info->mode == PIPE_PRIM_LINE_LOOP ||
-       info->mode == PIPE_PRIM_LINE_STRIP);
+      (info->mode == MESA_PRIM_LINES || info->mode == MESA_PRIM_LINE_LOOP ||
+       info->mode == MESA_PRIM_LINE_STRIP);
 
    return !lines && !fs->info.bifrost.uses_flat_shading;
 }
@@ -3428,7 +3428,7 @@ panfrost_get_position_shader(struct panf
    mali_ptr vs_ptr = batch->rsd[PIPE_SHADER_VERTEX];
 
    /* IDVS/triangle vertex shader */
-   if (vs_ptr && info->mode != PIPE_PRIM_POINTS)
+   if (vs_ptr && info->mode != MESA_PRIM_POINTS)
       vs_ptr += pan_size(SHADER_PROGRAM);
 
    return vs_ptr;
@@ -3923,7 +3923,7 @@ panfrost_draw(struct panfrost_batch *bat
 
    uint32_t dcd_flags0 = 0, dcd_flags1 = 0;
    pan_pack(&dcd_flags0, DCD_FLAGS_0, cfg) {
-      bool polygon = (u_reduced_prim(info->mode) == PIPE_PRIM_TRIANGLES);
+      bool polygon = (u_reduced_prim(info->mode) == MESA_PRIM_TRIANGLES);
 
       /*
        * From the Gallium documentation,
@@ -4023,7 +4023,7 @@ panfrost_draw(struct panfrost_batch *bat
    ceu_move32_to(b, ceu_reg32(b, 58), dcd_flags1);
 
    uint64_t primsize = 0;
-   panfrost_emit_primitive_size(ctx, info->mode == PIPE_PRIM_POINTS, 0,
+   panfrost_emit_primitive_size(ctx, info->mode == MESA_PRIM_POINTS, 0,
                                 &primsize);
    ceu_move64_to(b, ceu_reg64(b, 60), primsize);
 
@@ -4050,7 +4050,7 @@ panfrost_draw_vbo(struct pipe_context *p
    /* Do some common setup */
    struct panfrost_batch *batch = panfrost_get_batch_for_fbo(ctx);
 
-   bool points = (info->mode == PIPE_PRIM_POINTS);
+   bool points = (info->mode == MESA_PRIM_POINTS);
 
    if (unlikely(!panfrost_compatible_batch_state(batch, points))) {
       batch = panfrost_get_fresh_batch_for_fbo(ctx, "State change");
