diff --git a/meson.build b/meson.build
index 55d70dd..1394669 100644
--- a/meson.build
+++ b/meson.build
@@ -115,7 +115,7 @@ if dri_drivers.contains('auto')
     # TODO: PPC, Sparc
     if ['x86', 'x86_64'].contains(host_machine.cpu_family())
       dri_drivers = ['i915', 'i965', 'r100', 'r200', 'nouveau']
-    elif ['arm', 'aarch64'].contains(host_machine.cpu_family())
+    elif ['arm', 'aarch64', 'riscv64'].contains(host_machine.cpu_family())
       dri_drivers = ['r100', 'r200', 'nouveau']
     else
       error('Unknown architecture @0@. Please pass -Ddri-drivers to set driver options. Patches gladly accepted to fix this.'.format(
@@ -147,7 +147,7 @@ if gallium_drivers.contains('auto')
       gallium_drivers = [
         'r300', 'r600', 'radeonsi', 'nouveau', 'virgl', 'svga', 'swrast'
       ]
-    elif ['arm', 'aarch64'].contains(host_machine.cpu_family())
+    elif ['arm', 'aarch64', 'riscv64'].contains(host_machine.cpu_family())
       gallium_drivers = [
         'kmsro', 'v3d', 'vc4', 'freedreno', 'etnaviv', 'r300',
         'r600', 'radeonsi', 'nouveau',
@@ -206,7 +206,7 @@ if _vulkan_drivers.contains('auto')
   if system_has_kms_drm
     if host_machine.cpu_family().startswith('x86')
       _vulkan_drivers = ['amd', 'intel']
-    elif ['arm', 'aarch64'].contains(host_machine.cpu_family())
+    elif ['arm', 'aarch64', 'riscv64'].contains(host_machine.cpu_family())
       _vulkan_drivers = ['amd']
     else
       error('Unknown architecture @0@. Please pass -Dvulkan-drivers to set driver options. Patches gladly accepted to fix this.'.format(
